### ISSUE TITLE
Linker error on ValueTuple

### ISSUE THREAD

**Issue by kant2002** [dotnet/runtimelab#1356]
I try https://github.com/fsprojects/Avalonia.FuncUI/tree/master/src/Examples/Examples.GameOfLife with only upgrade to .NET 6.0 and adding `Microsoft.DotNet.ILCompiler` and receive this error.

```
Examples.GameOfLife.obj : error LNK2001: unresolved external symbol "const Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_5<Object__Object__Object__Object__Object>>::`vftable'" (??_7Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_5<Object__Object__Object__Object__Object>>@@6B@) [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
Examples.GameOfLife.obj : error LNK2001: unresolved external symbol "const Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_6<Object__Object__Object__Object__Object__Object>>::`vftable'" (??_7Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_6<Object__Object__Object__Object__Object__Object>>@@6B@) [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
Examples.GameOfLife.obj : error LNK2001: unresolved external symbol "const Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_7<Object__Object__Object__Object__Object__Object__Object>>::`vftable'" (??_7Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_7<Object__Object__Object__Object__Object__Object__Object>>@@6B@) [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
Examples.GameOfLife.obj : error LNK2001: unresolved external symbol "const Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_1<Object>>>::`vftable'" (??_7Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_1<Object>>>@@6B@) [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
Examples.GameOfLife.obj : error LNK2001: unresolved external symbol "const Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_2<Object__Object>>>::`vftable'" (??_7Boxed_S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_8<Object__Object__Object__Object__Object__Object__Object__S_P_CoreLib_System_ValueTuple_2<Object__Object>>>@@6B@) [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
bin\Release\net6.0\win-x64\native\Examples.GameOfLife.exe : fatal error LNK1120: 5 unresolved externals [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
packages\microsoft.dotnet.ilcompiler\6.0.0-rc.1.21373.2\build\Microsoft.NETCore.Native.targets(353,5): error MSB3073: The command ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\bin\Hostx64\x64\link.exe"  @"obj\Release\net6.0\win-x64\native\link.rsp"" exited with code 1120. [Avalonia.FuncUI\src\Examples\Examples.GameOfLife\Examples.GameOfLife.fsproj]
```

I have that issue before receiveing SEH like these #1316

**Comment by jkotas** [comment]
This may be a regressions from #1352 . cc @MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
Fixed in #1360.
